File: clients/sdk/frontend/typescript/src/types/message.ts
export type Message = {
  key: string;
  thread: string;
  author?: string;
  created_ts?: number;
  updated_ts?: number;
  body?: any;
  deleted?: boolean;
};

export type MessageCreateRequest = {
  body: any;
};

export type MessageUpdateRequest = {
  body: any;
};

export type MessageResponse = {
  message: Message;
};

File: clients/sdk/backend/python/src/progressdb/types/message.py
from typing import TypedDict, Any, Optional


class MessageType(TypedDict):
    key: str
    thread: str
    author: str
    created_ts: Optional[int]
    updated_ts: Optional[int]
    body: Optional[Any]
    deleted: Optional[bool]


class MessageCreateRequestType(TypedDict):
    body: Any  # Any JSON serializable data (object, string, number, etc.)


class MessageUpdateRequestType(TypedDict):
    body: Any  # Any JSON serializable data (object, string, number, etc.)